### PR TITLE
Release version 1.0.0

### LINKED ISSUE #1388: Release 1.0.0
Looking at the [milestone](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/milestone/16), it's about time we release version 1.0.0 (understatement intended).

Proposed target release date: **Monday July 2nd**.

I also think that it may be a good idea to do more regular (smaller) releases.

There are a couple of open issues/PRs which I'd like to see finished/merged before then:
- [x] PRs ~~#1374~~, ~~#1384~~, ~~#1386~~, ~~#1387~~ are IMO all good to go & ready to merge
- [ ] PR #1378 (previously #1233) still needs a little work, but not much. Would be great if @moorscode has some time this week to finish it off. I'll do a final review once the build passes.
    If @moorscode does not have time this week, this sniff can be moved to the 1.1.0 release.
- [x] PR #1366 - I need to do a last review & hope all is good. If so, I'll merge it ASAP ;-) /cc @grappler
    Update: review done, minor changes requested. 
- [ ] PR #1364 - If I remember correctly, that one still needs a little work on the token walking. Would be great if @grappler has some time this week to finish it off.
    If not, this sniff can be moved to the 1.1.0 release.
- [ ] PR #1349 and #1351 need a secondary review on the functions listed vs WP core. Any takers ? /cc @johnbillion 
- [ ] PR #1265 has an unresolved discussion. @JDGrimes What are your thoughts on this at this time ?
- [x] Issue #1368 - ~~I will open a PR for this shortly~~ PR #1390 addresses this.
- [ ] Issues #1218, #1272 (PHPCS 3.2.x annotations) - I'm still stuck on one last bit, but will take a decision on this soon and pull the PR fixing the issues I've found (I'm not guaranteeing completeness, but the PR should cover most cases).
    Update: PR #1407 takes care of the majority of this. Second PR upcoming.
- [ ] Issue #1254 is currently unresolved. It would be good to take a decision about this and if it would be decided to go ahead, implement this for the 1.0.0 version to prevent people having to rename the same sniff twice in rulesets/inline annotations. /cc @JDGrimes 
- [x] Create changelog for the 1.0.0 release - ~~volunteers welcome, if needs be, I can probably do this myself sometime this week~~ I'm working on this.

Opinions ? Additions ?

/cc @WordPress-Coding-Standards/wpcs-admins

### PR DESCRIPTION
**Pull request by jrfnl**
:warning: **DO NOT MERGE (YET)** :warning: 

PR for tracking changes for the 1.0.0 release. 
- [x] Add changelog for 1.0.0
- [ ] Merge this PR
- [ ] Add release tag & copy & paste the changelog to it
- [ ] Close the 1.0.0 milestone
- [x] Open a new milestone for the next release
- [ ] If any open PRs which were milestoned for 1.0.0 do not make it into the release, update their milestone.

Closes #1388

### CODE REVIEW

**Comment by GaryJones** [comment]
Has a conflict.

**Comment by jrfnl** [comment]
@GaryJones It shouldn't have as the same fix from 0.14.1 has also been included in the `develop` branch. Will do some tests locally, but I think this is the GH interface playing up for (yet another) imaginary merge conflicts.

**Comment by jrfnl** [comment]
@GaryJones Well, not GH interface, but Git itself in this case with an imaginary merge conflict. No worries, I'll try and sort it out in the changelog branch.

**Comment by jrfnl** [comment]
As PR #1431 should solve the remaining issue for the `1.0.0` release and PR #1430 will guard against the same happening ever again and PR #1423 (changelog) will solve the merge conflict, I'd like to ask you all to consider approving the release, providing, of course, that those three PRs will be approved and merged before release.

File: .phpcs.xml.dist [hunk 1]
```diff
@@ -0,0 +1,55 @@
+<?xml version="1.0"?>
+<ruleset name="WordPress Coding Standards">
+	<description>The Coding standard for the WordPress Coding Standards itself.</description>
+
+	<file>.</file>
+
+	<arg value="sp"/>
+	<arg name="extensions" value="php"/>
```
[the remark this answers is not in the document]
Comment (jrfnl) [reply .phpcs.xml.dist]: I'd be fine with those additions, but those can be added any time. They don't necessarily need to be included in this release, especially as they would/should only be added to the WPCS native custom ruleset.

File: composer.json [hunk 2]
```diff
@@ -25,7 +28,7 @@
 	},
 	"type"       : "phpcodesniffer-standard",
 	"scripts"    : {
-		"post-install-cmd": "\"vendor/bin/phpcs\" --config-set installed_paths ../../..",
-		"post-update-cmd" : "\"vendor/bin/phpcs\" --config-set installed_paths ../../.."
+		"post-install-cmd": "\"vendor/bin/phpcs\" --config-set installed_paths ../../..,../../phpcompatibility/php-compatibility",
+		"post-update-cmd" : "\"vendor/bin/phpcs\" --config-set installed_paths ../../..,../../phpcompatibility/php-compatibility"
```
Comment (GaryJones) [inline composer.json]: Let's keep this code dry:

```json
"scripts": {
	"install-codestandards": "\"vendor/bin/phpcs\" --config-set installed_paths ../../..,../../phpcompatibility/php-compatibility",
	"post-install-cmd": "@install-codestandards",
	"post-update-cmd": "@install-codestandards"
}
```

Comment (GaryJones) [reply composer.json]: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/1433

Comment (jrfnl) [reply composer.json]: I'd be fine with those changes, but they can be added any time.
They don't necessarily need to be included in this release, especially as it is only relevant for WPCS development anyway.

**Pull request merged**